### PR TITLE
WIP: API, CSV backend, dependencies and more

### PR DESCRIPTION
**Pull request by tebeka**
The PR is mostly to align code for working in parallel.

Known issues:
- Bug in server write - doesn't report last chunk
- Bug in server read - doesn't serve last package
- Go client write support untested

### CODE REVIEW

File: backends/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	return nil, fmt.Errorf("unknown type - %T", value)
+}
+
+// AppendNil appends an empty value to data
+func AppendNil(col frames.Column) error {
+	switch col.DType() {
+	case frames.IntType:
+		return col.Append(0)
+	case frames.FloatType:
+		return col.Append(0.0)
```
Comment (yaronha) [inline backends/utils/utils.go]: @tebeka should be math.NaN() ?

File: mapframe.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	return NewMapFrame(frameSlice)
+}
+
+// MapFrameMessage is over-the-wire frame data
+type MapFrameMessage struct {
+	Columns   []string                       `msgpack:"columns"`
+	SliceCols map[string]*SliceColumnMessage `msgpack:"slice_cols,omitempty"`
+	LabelCols map[string]*LabelColumnMessage `msgpack:"label_cols,omitempty"`
+}
+
```
Comment (yaronha) [inline mapframe.go]: @tebeka looks like some redundancy : 
1. Col names are in the col object 
2. Slice & Label can be part of the same list just different data types (Interface) 

also we need an indication which col is an index col (optional)

File: server/server.go [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+			ctx.Error(err.Error(), http.StatusBadRequest)
+			return
+		}
+	}
+
+	// TODO: Validate request
+
+	// TODO: We'd like to have a map of name->config of backends in configuration
+	backend, err := s.createBackend(request.Type)
+	if err != nil {
```
Comment (yaronha) [inline server/server.go]: @tebeka why do we create the backend every request ? we should create the backend on init

File: backends/csv/backend.go [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+		return nil, err
+	}
+
+	ca := &csvAppender{
+		writer:    file,
+		csvWriter: csv.NewWriter(file),
+		logger:    b.ctx.Logger,
+	}
+
+	return ca, nil
```
[the remark this answers is not in the document]
Comment (tebeka) [reply backends/csv/backend.go]: Yup. Fixed.

File: backends/utils/utils.go (hunk 1 above)
Comment (tebeka) [reply backends/utils/utils.go]: Done

File: server/server.go (hunk 3 above)
Comment (tebeka) [reply server/server.go]: Yeah. We talked about it and it - #7

**Pull request merged**